certsuite/harness.py

```diff
@@ -331,6 +332,7 @@ def install_marionette(device, version):
         logger.info("Installing marionette extension")
         try:
             marionette_install(version)
+            remove_after_mcts = True
```

Comment:
This won't be changing the global variable remove_after_mcts. Use global:
>>> a = "foo"
>>> def foo_bar():
...    a = "bar"
...
>>> a
'foo'
>>> foo_bar()
>>> a
'foo'
>>> def yes_bar():
...     global a
...     a = "bar"
...
>>> a
'foo'
>>> yes_bar()
>>> a
'bar'